packages/shorebird_cli/test/src/executables/shorebird_tools_test.dart

```diff
@@ -0,0 +1,154 @@
+import 'dart:io';
+
+import 'package:mocktail/mocktail.dart';
+import 'package:path/path.dart' as p;
+import 'package:scoped/scoped.dart';
+import 'package:shorebird_cli/src/executables/executables.dart';
+import 'package:shorebird_cli/src/logger.dart';
+import 'package:shorebird_cli/src/shorebird_env.dart';
+import 'package:shorebird_cli/src/shorebird_process.dart';
+import 'package:test/test.dart';
+
+import '../mocks.dart';
+
+void main() {
+  group('ShorebirdTools', () {
+    late File dartBinaryFile;
+    late Directory flutterDirectory;
+    late Directory tempDir;
+    late ShorebirdLogger logger;
+    late ShorebirdEnv shorebirdEnv;
+    late ShorebirdProcess process;
+    late ShorebirdProcessResult processResult;
+
+    R runWithOverrides<R>(R Function() body) {
+      return runScoped(
+        () => body(),
+        values: {
+          processRef.overrideWith(() => process),
+          shorebirdEnvRef.overrideWith(() => shorebirdEnv),
+          loggerRef.overrideWith(() => logger),
+          shorebirdToolsRef,
+        },
+      );
+    }
+
+    setUp(() {
+      tempDir = Directory.systemTemp.createTempSync();
+
+      flutterDirectory = Directory(p.join(tempDir.path, 'flutter'))
+        ..createSync();
+
+      dartBinaryFile = File(p.join(tempDir.path, 'dart'))..createSync();
+
+      processResult = MockProcessResult();
+      when(() => processResult.exitCode).thenReturn(0);
+      when(() => processResult.stdout).thenReturn('');
+      when(() => processResult.stderr).thenReturn('');
+
+      shorebirdEnv = MockShorebirdEnv();
+      when(() => shorebirdEnv.flutterDirectory).thenReturn(flutterDirectory);
+      when(() => shorebirdEnv.dartBinaryFile).thenReturn(dartBinaryFile);
+
+      process = MockShorebirdProcess();
+      when(
+        () => process.run(
+          any(),
+          any(),
+          workingDirectory: any(named: 'workingDirectory'),
+        ),
+      ).thenAnswer((_) async => processResult);
```

Comment:
We've historically instantiated the mocks in one block and configured them in another, e.g.:

```dart
      processResult = MockProcessResult();
      shorebirdEnv = MockShorebirdEnv();

      when(() => processResult.exitCode).thenReturn(0);
      when(() => processResult.stdout).thenReturn('');
      when(() => processResult.stderr).thenReturn('');

      when(() => shorebirdEnv.flutterDirectory).thenReturn(flutterDirectory);
      when(() => shorebirdEnv.dartBinaryFile).thenReturn(dartBinaryFile);
```

Not sure that's the Right™ way to do it, but we should be consistent